pkg/api/admin/openshiftcluster_validatestatic.go

```diff
@@ -29,9 +29,37 @@ func (sv openShiftClusterStaticValidator) validateDelta(oc, current *OpenShiftCl
 		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodePropertyChangeNotAllowed, err.Target, err.Message)
 	}
 
-	if !(oc.Properties.MaintenanceTask == "" || oc.Properties.MaintenanceTask == MaintenanceTaskEverything || oc.Properties.MaintenanceTask == MaintenanceTaskOperator || oc.Properties.MaintenanceTask == MaintenanceTaskRenewCerts) {
+	if invalidMaintenanceTask(oc.Properties.MaintenanceTask) {
 		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, "properties.maintenanceTask", "Invalid enum parameter.")
 	}
 
+	if invalidMaintenanceState(oc.Properties.MaintenanceState) {
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, "properties.maintenanceState", "Invalid enum parameter.")
+	}
+
 	return nil
 }
+
+func invalidMaintenanceTask(task MaintenanceTask) bool {
+	if task == "" ||
+		task == MaintenanceTaskEverything ||
+		task == MaintenanceTaskOperator ||
+		task == MaintenanceTaskRenewCerts ||
+		task == MaintenanceTaskStateUpdate {
+		return false
+	}
+
+	return true
```

Comment:
I know this is still a draft, but you should use a switch here, that will be easier